### PR TITLE
Update api

### PR DESCRIPTION
**Pull request by gferraro**
use deviceID for authenticate_device where available
save deviceID from registration response if available

### CODE REVIEW

File: app/src/main/java/nz/org/cacophony/birdmonitor/Server.java [hunk 1]
```diff
@@ -91,10 +91,16 @@ public static boolean login(Context context) {
 
             String loginUrl = prefs.getServerUrl() + LOGIN_URL;
 
-            RequestBody requestBody = new FormBody.Builder()
-                    .add("devicename", devicename)
-                    .add("password", devicePassword)
-                    .build();
+            FormBody.Builder builder = new FormBody.Builder();
+            if(deviceID > 0) {
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/birdmonitor/Server.java]: space after `if`

File: app/src/main/java/nz/org/cacophony/birdmonitor/Server.java [hunk 2]
```diff
@@ -91,10 +91,16 @@ public static boolean login(Context context) {
...
+                builder.add("deviceID", Long.toString(deviceID));
+            }else {
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/birdmonitor/Server.java]: space after `}`

File: app/src/main/java/nz/org/cacophony/birdmonitor/Server.java [hunk 3]
```diff
@@ -251,16 +257,18 @@ public static void registerDevice(final String group, final String deviceName, f
             }
 
             if (response.isSuccessful()) {
-
+                long deviceID = 0;
+                //makes backwards compatible
+                if(responseJson.has("id")) {
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/birdmonitor/Server.java]: as above

File: app/src/main/java/nz/org/cacophony/birdmonitor/Server.java [hunk 4]
```diff
@@ -251,16 +257,18 @@ public static void registerDevice(final String group, final String deviceName, f
...
+                    deviceID = responseJson.getLong("id");
+                }else {
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/birdmonitor/Server.java]: as above

File: app/src/main/java/nz/org/cacophony/birdmonitor/Util.java [hunk 5]
```diff
@@ -234,14 +234,18 @@ public static File getRecordingsOrNotesFolder(Context context, String recordings
...
-            return "";
+            return 0;
         }
 
         String webTokenBody = Util.decoded(webToken);
         JSONObject jObject = new JSONObject(webTokenBody);
-        return jObject.getString("id");
+        try {
+            return Long.parseLong(jObject.getString("id"));
+        }catch (NumberFormatException ex){
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/birdmonitor/Util.java]: space after `}` and `)`

Comment (gferraro) [reply app/src/main/java/nz/org/cacophony/birdmonitor/Util.java]: Sorry need to pay more attention to these

**Pull request merged**